test/PlatformFunctions.spec.ts

```diff
@@ -0,0 +1,194 @@
+        const crypto = require('crypto');
+        let randomUUID$;
+        let called;
+        beforeEach(() => {
+          // if it's available, mock it and ensure it's called
+          // otherwise, skip this whole operation
+          if (isNodeEnvironment && crypto.randomUUID) {
+            called = false;
+            randomUUID$ = crypto.randomUUID;
+            crypto.randomUUID = (): string => {
+              called = true;
+              return 'no, YOU you id';
+            };
+          }
+        });
+        afterEach(() => {
+          if (randomUUID$) {
+            crypto.randomUUID = randomUUID$;
+          }
+        });
+        it('is called if available', () => {
+          if (randomUUID$) {
+            expect(platformFunctions.uuid4()).to.equal('no, YOU you id');
+            expect(called).to.equal(isNodeEnvironment);
+          }
+        });
+
+        it('returns a valid v4 UUID without it', () => {
+          crypto.randomUUID = null;
+          expect(platformFunctions.uuid4()).to.match(
+            // regex from https://createuuid.com/validator/, specifically for v4
+            /^[0-9a-f]{8}-[0-9a-f]{4}-4[0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}$/i
+          );
+        });
+      });
+
+      it('should return a well-formatted v4 UUID', () => {
+        expect(platformFunctions.uuid4()).to.match(
+          // regex from https://createuuid.com/validator/, specifically for v4
+          /^[0-9a-f]{8}-[0-9a-f]{4}-4[0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}$/i
+        );
+        // further test: could spy on crypto.randomUUID to ensure it's being used, if available
+        // whether that's useful is a race between using jest/sinon for these tests and dropping support for node < 14
+      });
+    });
+
+    describe('secureCompare with crypto.timingSafeEqual', () => {
+      it('returns true given two equal things', () => {
+        expect(platformFunctions.secureCompare('potato', 'potato')).to.equal(
+          true
+        );
+      });
+
+      it('returns false given two unequal things', () => {
+        expect(platformFunctions.secureCompare('potato', 'tomato')).to.equal(
+          false
+        );
+      });
+
+      it('returns false given things of unequal lengths', () => {
+        expect(platformFunctions.secureCompare('potat', 'tomato')).to.equal(
+          false
+        );
+      });
+
+      it('throws an error if not given two things to compare', () => {
+        expect(() => {
+          platformFunctions.secureCompare('potato');
+        }).to.throw();
+      });
+    });
+
+    describe('secureCompare without crypto.timingSafeEqual', () => {
+      const crypto = require('crypto');
+      let timingSafeEqual$;
+      beforeEach(() => {
+        // if it's available, mock it and ensure it's called
+        // otherwise, skip this whole operation
+        if (isNodeEnvironment && crypto.timingSafeEqual) {
+          timingSafeEqual$ = crypto.timingSafeEqual;
```

Comment:
Do we still need this logic?

Reply:
This logic is still necessary, since `crypto.timingSafeEqual` is used in `secureCompare` in Node environments and we want to ensure that secure compare still works when the `crypto` operation is not available.